go-controller/pkg/ovn/master.go

```diff
@@ -538,29 +538,85 @@ func (oc *Controller) SetupMaster(masterNodeName string) error {
 	return nil
 }
 
-func addNodeLogicalSwitchPort(logicalSwitch, portName, portType, addresses, options string) (string, error) {
-	stdout, stderr, err := util.RunOVNNbctl("--", "--may-exist", "lsp-add", logicalSwitch, portName,
-		"--", "lsp-set-type", portName, portType,
-		"--", "lsp-set-options", portName, options,
-		"--", "lsp-set-addresses", portName, addresses)
+func addNodeLogicalSwitchPort(ovnNBClient goovn.Client, logicalSwitch, portName, portType, addresses, options string) (string, error) {
+	var cmds []*goovn.OvnCommand
+	var goOvnOptions map[string]string
+
+	lsp, err := ovnNBClient.LSPGet(portName)
+	if err != nil && err != goovn.ErrorNotFound && err != goovn.ErrorSchema {
+		return "", fmt.Errorf("unable to get the lsp: %s from the nbdb: %s", portName, err)
+	}
+
+	// Handle --may-exist by returning the UUID of the existing LSP
+	if lsp != nil {
```

Comment:
If exists, you may want to proceed and set address and the other parameters

Reply:
Done..